### DISCUSSION TITLE
Split yaml from appdata

### DISCUSSION THREAD

**Discussion by BrokenSpring** [louislam/dockge#516] [Ask for Help] [answered]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### 📝 Describe your problem

Hello All
I absolutely love Dockge and its capabilities, but i have an issue.
I have chosen OMV7 as my chosen NAS OS. And it uses docker compose.
But it splits the yaml files in separate folders in its own Compose folder, and the same for appdata
so i have 
Root/compose/uptime-kuma/uptime-kuma.yaml + environment variables
Root/appdata/uptime-kuma/app files

Is there a way to make Dockge work with the compose file and app files split in different folders?

### 🐻 Dockge Version

3.8

### 💻 Operating System and Arch

OpenMediaVault 7

### 🌐 Browser

Firefox 125.0.3

### 🐋 Docker Version

Docker compose  7.2

**Comment by trythatagain** [comment]
I too use OMV7, but tend to use the compose plugin, until Dockge has a global .env built in (I know there is a workaround, but I want it built in) .
So I'm not sure how you have it all setup, but I'll try and  set out how I have it setup. I do use a separate folder for compose (stacks) and appdata. Setup Dockge as per instructions, so you have your compose(stacks) folder setup.

I then just use something like the following in my compose files, I just checked OMV and the 'appdata' shared folder in settings is optional, so I don't actually use it, I just place my appdata folder as a volume in all my compose files. Which means you can have more than 1 appdata folder.

 
```    
volumes:
     - /srv/dev-disk-by-uuid-2b9b3914-0f0d-4737-9638-etc/appdata/diun:/data
```

**Reply by BrokenSpring** [reply]
Hmmm it could be i am setting it up wrong then.?
In the OMV compose i have 2 separate folders setup.
one for compose files, and one for appdata.
And in those i have folders for each docker by name.
So like the below

    /srv/dev-disk-by-uuid-2b9b3914-0f0d-4737-9638-etc/appdata/netdata
    /srv/dev-disk-by-uuid-2b9b3914-0f0d-4737-9638-etc/compose/netdata

but when i specify in Dockge compose one or the other, it sees the dockers but cant manage them.

So is there a way i can specify diffrent paths for the compose and appdata. So i can use Dockge or the compose in OMV
or would i need to combine the folders so the compose files are in the same folder as appdata?

**Comment by trythatagain** [comment, marked as answer]
Right I think I can see whats wrong, I'll try and explain it as best as I can, if I don't come across right just please ask.

OMV and Dockge have different compose file names - OMV 'container.yml' and 'container.env' + the 'global.env', Dockge 'compose.yml' and '.env' (I did ask compose plugin maintainer if he's change to default name, and he declined, which is fair enough)

So because we set the below volume, Dockge can see containers that are up.
```
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
```
In my setup Dockge will say they are active, but thats it, I can't 'down' it, and thats because of the compose file names.

I did write a simple script that will go thru the 'compose' folder and copy 'container.yml' to 'compose.yml' then take 'global.env' and 'container.env' file and copy it to '.env' file. This then allows me to mess about with Dockge, I think I might have either posted it or asked if anyone wanted me to, and no-one replied so I lost interest :( 

Just remember everything in OMV uses a database, so all the compose files that OMV saved have to be edited from within OMV for changes to stick. So if you changed the 'compose.yml' or '.env' with Dockge and still want it to show in OMV you'll need to edit in OMV or I change in OMV and then run my script.

But when Dockge has global .env, I might ditch the compose plugin as it does loads I don't even look at.

**Reply by BrokenSpring** [reply]
Ah you are a star. That makes sense.

So its not so much the diff directories as the names are diffrent so it can never see the file.
I really like the omv plugin, but love the extra stuff you can see in one go in Dockge.
So really i need to do one or other, same as choosing to use portainer or omv compose.
decisions decisions. which to pick.
Thank you kindly for the help and explanation.

**Comment by trythatagain** [comment]
Thats homelabing for you, always something else shiny and new around the corner :)
